packages/dataviews/src/dataviews-layouts/picker-grid/index.tsx

```diff
@@ -306,8 +306,9 @@ function ViewPickerGrid< Item >( {
 	 */
 	const size = '900px';
 
-	const groupField = view.groupByField
-		? fields.find( ( f ) => f.id === view.groupByField )
+	const groupFieldId = view.groupBy?.field || view.groupByField;
```

Comment:
Besides whether we do these groupBy changes here or separately, is there a plan to remove this fallback in the future? Should we consider it a breaking change and also update core's usages in one PR?

If we kept the fallback, I guess we should add comments about when to remove this.